### ISSUE TITLE
cram: Required number of data containers

### ISSUE THREAD

**Issue by zaeleus** [samtools/hts-specs#636]
This is regarding [_CRAM format specification (version 3.0)_](https://github.com/samtools/hts-specs/blob/ef58a8c058a06585c81bb33470b1dcaabd5a8f91/CRAMv3.pdf) (2021-10-15).

§ 5 "File structure" states that a CRAM file has "one or more data containers":

> A CRAM file consists of a fixed length file definition, followed by a CRAM header container, then one or more data containers, and finally a special end-of-file container.

Should this be _zero_ or more data containers? Otherwise, this definition invalidates `test/cram/3.0/passed/{0001_empty_eof,0100_header1,0101_header2}.cram`, which have no data containers.

**Comment by jkbonfield** [comment]
Yes, good catch; empty files are valid.  It's a trivial amendment and thanks for the report.  I'll make a PR.
